### PR TITLE
Document passive option

### PR DESCRIPTION
**Pull request by sherakama**
# Description

Document support for passive option in `node-saml`

Re: https://github.com/node-saml/node-saml/blob/master/src/saml.ts#L135

# Checklist:

- Issue Addressed: []
- Link to SAML spec: []
- Tests included? []
- Documentation updated? [x]

### CODE REVIEW

**Comment by cjbarth** [comment]
Thanks for adding that documentation. Do you happen to have a reference for the part of the SAML spec that refers to? That feature was added a while ago by someone else. I'd like to refresh my understanding of your description before approving :)

**Comment by sherakama** [comment]
There is this https://docs.oracle.com/en/middleware/idm/access-manager/12.2.1.3/aiaag/introducing-identity-federation-oracle-access-management.html

I'll see about something more specific to the SAML 2.0 spec.

**Comment by sherakama** [comment]
https://www.oasis-open.org/committees/download.php/8958/sstc-saml-implementation-guidelines-draft-01.pdf

Page 33 - Section 7.2.1

**Comment by cjbarth** [comment]
I'm having trouble finding actual documentation on the `IsPassive` flag that `passport-saml` actually sets. The best I've found so far discusses largely what you've said, but still no references to it being part of a spec: https://lists.oasis-open.org/archives/saml-dev/201605/msg00004.html


> Hi,
> 
> theoretically such seemless authentication is still within the bounds of the isPassive definition. Such authentication methods could probably involve: * kerberos (as long as the 401 page is not being displayed to the end-user after failed login)
> 
> * remember me feature (a.k.a persistent cookie)
> 
> Cert based authn is also similar, but with that the browser will ask for the password of the private key, not sure if that qualifies as passive authn.
> 
> 
> cheers,
> Peter
> 
> 2016. 05. 10. 12:17 keltezéssel, Chiranga Alwis írta:
> 
> > Hi Dieter,
> > 
> > first of all thanks for the answer.
> > 
> > In addition, to what you have mentioned I came across this alternative
> > usage of this property (from another source) which enables the Service
> > Provider to ask the IdP to basically "authenticate this user only if you
> > can do it without the having the user involved". This seems to map with
> > the spec definition provided for the 'IsPassive' property under
> > discussion. Is this a valid usage and what does it accurately mean in a
> > real-world scenario?
> > 
> > 
> > 
> > > On Tue, May 10, 2016 at 1:18 PM, Dieter Maurer <dieter@handshake.de

File: README.md [hunk 1]
```diff
@@ -165,6 +165,7 @@ export interface Profile {
 - `racComparison`: Requested Authentication Context comparison type. Possible values are 'exact','minimum','maximum','better'. Default is 'exact'.
 
 - `forceAuthn`: if set to true, the initial SAML request from the service provider specifies that the IdP should force re-authentication of the user, even if they possess a valid session.
+- `passive`: if set to true, the initial SAML request from the service provider specifies that the IdP should prevent visible user interaction if possible.
```
Comment (cjbarth) [inline README.md:168]: Based on the spec, "if possible" isn't true. The spec says the IdP "must not". Is there a different resource where you found that it obeys this less strict implementation?

**Comment by cjbarth** [comment]
Here it is on page 49: https://docs.oasis-open.org/security/saml/v2.0/saml-core-2.0-os.pdf

> ForceAuthn [Optional]
> A Boolean value. If "true", the identity provider MUST authenticate the presenter directly rather than
> rely on a previous security context. If a value is not provided, the default is "false". However, if both
> ForceAuthn and IsPassive are "true", the identity provider MUST NOT freshly authenticate the
> presenter unless the constraints of IsPassive can be met.
> IsPassive [Optional]
> A Boolean value. If "true", the identity provider and the user agent itself MUST NOT visibly take control
> of the user interface from the requester and interact with the presenter in a noticeable fashion. If a
> value is not provided, the default is "false".

File: README.md (hunk 1 above)
Comment (sherakama) [reply README.md:168]: > Based on the spec, "if possible" isn't true. The spec says the IdP "must not". Is there a different resource where you found that it obeys this less strict implementation?

I can't say I have for SAML. I am happy to change this language.

**Comment by sherakama** [comment]
@cjbarth,

Thanks for reviewing this and finding the documentation. I have updated the language. I hope this helps.

**Review by cjbarth — DISMISSED** [review]

**Comment by cjbarth** [comment]
Please run `npm run prettier-format` to correct the code formatting.

**Comment by sherakama** [comment]
Ok.

**Review by cjbarth — APPROVED** [review]

**Comment by sherakama** [comment]
Thanks @cjbarth

**Pull request merged**